garage/tf/policies/base2.py

```diff
@@ -88,6 +94,25 @@ def terminate(self):
         """Clean up operation."""
         pass
 
+    def get_trainable_vars(self, scope=None):
```

Comment:
why even have `scope=`?

Reply:
We can a situation that we only have one policy object, but there is also a target policy. We also need a way to get the variables under the target policy.
For example in DDPG
```python
self.target_policy_output = self.policy.dist_info_sym(target_obs_ph, name="target_policy")

policy_init_ops, policy_update_ops = get_target_ops(
                    self.policy.get_global_vars(),
                    self.policy.get_global_vars("target_policy"), self.tau)
```

Reply:
this example is really confusing to me. if there's a regular policy and a target policy, then why don't i have `self.policy` and `self.target_policy`?

Reply:
The current design without model is that the policy object contained the network structure, and calling `build_net()` gave us a new network. It reuses the network structure.

With model, we will need to pass two policies to the algorithm in this case.

Can we have something that also allows us to reuse the network structure (model)? Instead of passing n policies, we then just need one policy and build n times.

Reply:
Do we need this function with model?

Reply:
Without this function, algorithm would need to have n policy object. Or we can do `model.copy()` when we need duplicate (e.g. target network).

Reply:
what does `n` equal?

Reply:
Usually n would be 2 (trainable policy and target policy), e.g. in DDPG and DQN. If we also do the same for QFunction, it would be 2 policies and 2 QF. For TD3, it would be 2 policies and 4 QF.

Reply:
Then I think we should copy the model. If an algorithm needs to maintain multiple copies of parameters, it needs multiple copies of the model. 1:1.